internal/services/network/network_manager_verifier_workspace_resource.go

```diff
@@ -0,0 +1,264 @@
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			networkManagerId := ipampools.NewNetworkManagerID(id.SubscriptionId, id.ResourceGroupName, id.NetworkManagerName).ID()
+			schema := ManagerVerifierWorkspaceResourceModel{
+				Name:             id.VerifierWorkspaceName,
+				NetworkManagerId: networkManagerId,
+			}
+
+			if model := resp.Model; model != nil {
+				schema.Location = location.Normalize(model.Location)
+				schema.Tags = tags.Flatten(model.Tags)
+
+				if props := model.Properties; props != nil {
+					schema.Description = pointer.From(props.Description)
+				}
+			}
+
+			return metadata.Encode(&schema)
+		},
+	}
+}
+
+func (r ManagerVerifierWorkspaceResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.VerifierWorkspaces
+
+			id, err := verifierworkspaces.ParseVerifierWorkspaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model ManagerVerifierWorkspaceResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			parameters := verifierworkspaces.VerifierWorkspaceUpdate{
+				Properties: &verifierworkspaces.VerifierWorkspaceUpdateProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				parameters.Tags = tags.Expand(model.Tags)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				parameters.Properties.Description = pointer.To(model.Description)
+			}
+
+			if _, err := client.Update(ctx, *id, parameters); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (r ManagerVerifierWorkspaceResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Network.VerifierWorkspaces
+
+			id, err := verifierworkspaces.ParseVerifierWorkspaceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			// https://github.com/Azure/azure-rest-api-specs/issues/31688
+			if err := resourceVerifierWorkspaceWaitForDeleted(ctx, *client, *id); err != nil {
+				return fmt.Errorf("waiting for %s to be deleted: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceVerifierWorkspaceWaitForDeleted(ctx context.Context, client verifierworkspaces.VerifierWorkspacesClient, id verifierworkspaces.VerifierWorkspaceId) error {
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("internal error: context had no deadline")
+	}
+
+	state := &pluginsdk.StateChangeConf{
```

Comment:
We shouldn't be using `StateChangeConf` for LROs anymore - can you please create a custom poller for this? Here is an [example](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/maps/custompollers/maps_account_poller.go) to help you get started.

Reply:
updated, a custom poller is added